document/core/exec/instructions.rst

```diff
@@ -974,97 +945,111 @@ Table Instructions
 :math:`\TABLECOPY`
 ..................
 
-1. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+1. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
 
-2. Pop the value :math:`\I32.\CONST~cnt` from the stack.
+2. Assert: due to :ref:`validation <valid-table.copy>`, :math:`F.\AMODULE.\MITABLES[0]` exists.
 
-3. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+3. Let :math:`\X{ta}` be the :ref:`table address <syntax-tableaddr>` :math:`F.\AMODULE.\MITABLES[0]`.
 
-4. Pop the value :math:`\I32.\CONST~src` from the stack.
+4. Assert: due to :ref:`validation <valid-table.copy>`, :math:`S.\STABLES[\X{ta}]` exists.
 
-5. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+5. Let :math:`\X{tab}` be the :ref:`table instance <syntax-tableinst>` :math:`S.\STABLES[\X{ta}]`.
 
-6. Pop the value :math:`\I32.\CONST~dst` from the stack.
+6. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-7. If :math:`cnt = 0`, then:
+7. Pop the value :math:`\I32.\CONST~n` from the stack.
 
-   a. Return.
+8. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-8. If :math:`cnt = 1`, then:
+9. Pop the value :math:`\I32.\CONST~s` from the stack.
 
-   a. Push the value :math:`\I32.\CONST~dst` to the stack.
+10. Assert: due to :ref:`validation <valid-table.copy>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
 
-   b. Push the value :math:`\I32.\CONST~src` to the stack.
+11. Pop the value :math:`\I32.\CONST~d` from the stack.
 
-   c. Execute the instruction :math:`\TABLEGET`.
+12. If :math:`s + n` is larger than the length of :math:`\X{tab}.\TIELEM` or :math:`d + n` is larger than the length of :math:`\X{tab}.\TIELEM`, then:
 
-   d. Execute the instruction :math:`\TABLESET`.
+    a. Trap.
 
-   e. Return.
+13. If :math:`n = 0`, then:
 
-9. If :math:`dst <= src`, then:
+   a. Return.
+
+14. If :math:`d \leq s`, then:
+
+   a. Push the value :math:`\I32.\CONST~d` to the stack.
```

Comment:
I believe you meant to load from the source offset here.

Reply:
Done.